include/coap3/coap_mutex_internal.h

```diff
@@ -61,16 +64,28 @@ typedef int coap_mutex_t;
 #elif defined(WITH_CONTIKI)
 /* Contiki does not have a mutex API, used as single thread */
 typedef int coap_mutex_t;
-#define COAP_MUTEX_INITIALIZER 0
+#define COAP_MUTEX_DEFINE(_name)			\
+	static coap_mutex_t _name
 #define coap_mutex_lock(a) *(a) = 1
 #define coap_mutex_trylock(a) *(a) = 1
 #define coap_mutex_unlock(a) *(a) = 0
 
-#else /* !WITH_CONTIKI && !WITH_LWIP && !RIOT_VERSION && !HAVE_PTHREAD_H && !HAVE_PTHREAD_MUTEX_LOCK */
+#elif defined(__ZEPHYR__)
+#include <zephyr/sys/mutex.h>
+
+typedef struct k_mutex coap_mutex_t;
+#define COAP_MUTEX_DEFINE(_name)			\
+	static SYS_MUTEX_DEFINE(_name)
+#define coap_mutex_lock(a) sys_mutex_lock(a, K_FOREVER)
+#define coap_mutex_trylock(a) sys_mutex_lock(a, K_NO_WAIT)
+#define coap_mutex_unlock(a) sys_mutex_unlock(a)
+
+#else /* !__ZEPYR__ && !WITH_CONTIKI && !WITH_LWIP && !RIOT_VERSION && !HAVE_PTHREAD_H && !HAVE_PTHREAD_MUTEX_LOCK */
 /* define stub mutex functions */
 #warning "stub mutex functions"
 typedef int coap_mutex_t;
-#define COAP_MUTEX_INITIALIZER 0
+#define COAP_MUTEX_DEFINE(_name)			\
+	static coap_mutex_t _name
```

Comment:
So that it is properly initialized, I think I would like to see this as
```
    static coap_mutex_t _name = 0
````
even though it does not really make any difference.